src/xcode/ENA/ENA/Source/Services/Risk/Calculation/__tests__/RiskCalculationTests.swift

```diff
@@ -332,6 +332,29 @@ final class RiskCalculationTests: XCTestCase {
 		XCTAssertTrue(risk?.riskLevelHasChanged == false)
 	}
 
+	func testCalculateRisk_IncreasedOverridesUnknownOutdated() {
```

Comment:
If I understand correctly, for `.increased` to override `.unknownOutdated`, we need all three things to be `true`:

- Date of last exposure detection is older than 48 hours.
- Exposure detection returned a summary which will lead to `.increased` risk.
- `.unknownInitial` state is present, with tracing being active less than 24 hours.

According to these rules, I think we need to adjust this unit test in the following way:
- Change `dateLastExposureDetection` parameter to something like `Date().addingTimeInterval(.init(hours: -25))`, to nudge it over the limit defined in the `RiskProvidingConfiguration` on line 337
- Change `activeTracing` parameter to something like `.init(interval: 2 * 3600)` so that `.unknownInitial` condition triggers.

I think then this unit test covers the three conditions listed above (it still passes with these changes though 👍) We're getting super deep into edge cases here but that's good 😀

Is my thinking correct?

Reply:
so I understand it this way - `.increased` overrides `.unknownOutdated` in all cases, except one case.

Here is the case:

- `.unknownOutdated` state is present
- `.increased` is present
- `.unknownInitial` is **NOT** present

It's Scenario 3 from the document. So the result will be `.unknownOutdated` state.

In all other cases the result will be `.increased` state, for example:

1) 
- `.unknownOutdated` state is present
- `.increased` is present
- `.unknownInitial` is present
2)
- `.unknownOutdated` state is **NOT** present
- `.increased` is present
- `.unknownInitial` is **NOT** present
3)
- `.unknownOutdated` state is **NOT** present
- `.increased` is present
- `.unknownInitial` is present

In all these 3 cases the result will be `.increased` state.

So this unit test covers 2) case and you described 1) case.

:)

Reply:
I added additional 2 unit tests for it